src/Nethermind/Nethermind.Merge.Plugin/Data/ExecutionPayloadForDebugRpc.cs

```diff
@@ -0,0 +1,44 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core.Crypto;
+using Nethermind.Merge.Plugin.Data;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+
+namespace Nethermind.JsonRpc.Data
+{
+    [JsonPolymorphic(TypeDiscriminatorPropertyName = "$type")]
+    [JsonDerivedType(typeof(ParamsV1), typeDiscriminator: "v1")]
+    [JsonDerivedType(typeof(ParamsV3), typeDiscriminator: "v3")]
+    [JsonDerivedType(typeof(ParamsV4), typeDiscriminator: "v4")]
+    public class Params;
+    public class ParamsV1(ExecutionPayload executionPayload) : Params
+    {
+        public ExecutionPayload ExecutionPayload { get; set; } = executionPayload ?? throw new ArgumentNullException(nameof(executionPayload));
+    }
```

Comment:
Apply to others too:

```suggestion
    public record class ParamsV1(ExecutionPayload ExecutionPayload) : Params;
```